src/OrchardCore.Modules/OrchardCore.Sitemaps/Controllers/SitemapController.cs

```diff
@@ -86,7 +86,7 @@ public async Task<IActionResult> Index(string sitemapId, CancellationToken cance
 
                     document.Declaration = new XDeclaration("1.0", "utf-8", null);
 
-                    var stream = new MemoryStream();
+                    using var stream = MemoryStreamFactory.GetStream();
```

Comment:
I'm not using the Site Map feature, so I can't say whether this is actually a problem, but this looks wrong. With this change, the stream is disposed as it's being returned for use later (by the File result).

Reply:
it look suspicious to me too. Feel free to test it out, and submit an issue and PR.

Reply:
This is indeed a problem. I am fixing it.